python/cudf/cudf/tests/test_parquet.py

```diff
@@ -518,6 +526,186 @@ def test_parquet_reader_multiple_files(tmpdir, src):
     assert_eq(expect, got)
 
 
+def test_parquet_reader_list_basic(tmpdir):
+    expect = pd.DataFrame({"a": [[[1, 2], [3, 4]], None, [[5, 6], None]]})
+    fname = tmpdir.join("test_parquet_reader_list_basic.parquet")
+    expect.to_parquet(fname)
+    assert os.path.exists(fname)
+    got = cudf.read_parquet(fname)
+    assert_eq(expect, got)
+
+
+def test_parquet_reader_list_table(tmpdir):
+    expect = pd.DataFrame(
+        {
+            "a": [[[1, 2], [3, 4]], None, [[5, 6], None]],
+            "b": [[None, None], None, [None, None]],
+            "c": [[[1, 2, 3]], [[None]], [[], None]],
+            "d": [[[]], [[None]], [[1, 2, 3], None]],
+            "e": [[["cows"]], [["dogs"]], [["cats", "birds", "owls"], None]],
+        }
+    )
+    fname = tmpdir.join("test_parquet_reader_list_table.parquet")
+    expect.to_parquet(fname)
+    assert os.path.exists(fname)
+    got = cudf.read_parquet(fname)
+    assert_eq(expect, got, check_dtype=False)
+
+
+def int_gen(first_val, i):
+    return int(i + first_val)
+
+
+strings = [
+    "cats",
+    "dogs",
+    "cows",
+    "birds",
+    "fish",
+    "sheep",
+    "owls",
+    "bears",
+    "ants",
+]
+
+
+def string_gen(first_val, i):
```

Comment:
A small docstring explaining what this function does would be very helpful. Ditto for `list_gen` below